crates/nu-command/src/core_commands/while_.rs

```diff
@@ -46,6 +48,12 @@ impl Command for While {
         let block: Block = call.req(engine_state, stack, 1)?;
 
         loop {
+            if let Some(ctrlc) = &engine_state.ctrlc {
+                if ctrlc.load(Ordering::SeqCst) {
+                    break;
```

Comment:
Yeah as `while` doesn't return the content regularly on the pipeline, just breaking should be fine.